lib/filters/split.py

```diff
@@ -0,0 +1,33 @@
+"""
+Copyright 2016 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from ansible.errors import AnsibleFilterError
+
+
+def split(data, separator=',', maxsplit=-1):
+    try:
+        return data.split(separator, int(maxsplit))
+    except Exception as ex:
+        raise AnsibleFilterError(
+            "Failed to split {} with {} due to:\n{}".format(data, separator, ex)
+        )
+
+
+class FilterModule(object):
+    def filters(self):
+        return {
+            'split': split
+        }
```

Comment:
I'm not sure I see why we need to introduce a `split` filter?  You can just call the `split` method on strings, e.g.:

    {% for thing in list_of_things.split(',') %}